### PR TITLE
Tas 75 handle auth

### PR DESCRIPTION
**Pull request by Kepmon**
This branch was created in order to handle all auth-related stuff, including creating layout as well as adding the functionality.

At the moment, the layout was created, however, it still requires some small improvements.

Next step: Handle user auth using Firebase Authentication.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,14 +9,23 @@
     "test:unit": "vitest",
     "build-only": "vite build",
     "type-check": "vue-tsc --noEmit -p tsconfig.vitest.json --composite false",
-    "lint": "eslint . --ext .vue,.js,.jsx,.cjs,.mjs,.ts,.tsx,.cts,.mts --fix --ignore-path .gitignore"
+    "lint": "eslint . --ext .vue,.js,.jsx,.cjs,.mjs,.ts,.tsx,.cts,.mts --fix --ignore-path .gitignore",
```
Comment (aleksanderwalczuk) [inline package.json]: Remove fix param from this one

File: src/components/BoardsColumn.vue [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    'bg-blue-500': props.columns.indexOf(column) % 3 === 1,
+    'bg-green-400': props.columns.indexOf(column) % 3 === 2
+  })
+})
+
+const clickedTitle: Ref<null | Task['title']> = ref(null)
+const changeClickedTask = (e: Event) => {
+  const titleOfClickedTask = (e.currentTarget as HTMLElement)?.querySelector(
+    'p'
+  )?.textContent
```
Comment (aleksanderwalczuk) [inline src/components/BoardsColumn.vue]: Instead of native event on the parent component it should be an emit handler

File: src/components/Dialogs/BoardDialog.vue [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import type { BoardColumn } from '../../api/boardsTypes'
+import DialogsTemplate from './DialogsTemplate.vue'
+import InputText from '../shared/InputText.vue'
+import MultiOption from './elements/MultiOption.vue'
+import TheButton from '../../components/shared/TheButton.vue'
+
+defineProps<{
+  action: 'add' | 'edit'
+  selectedMultiOptionItems: string[] | BoardColumn['name'][]
+  closeDialog: () => void
```
Comment (aleksanderwalczuk) [inline src/components/Dialogs/BoardDialog.vue]: Why prop? Use emit instead

File: src/components/Dialogs/ConfirmationDialog.vue [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+</template>
+
+<script setup lang="ts">
+import DialogsTemplate from './DialogsTemplate.vue'
+import TheButton from '../shared/TheButton.vue'
+import { computed } from 'vue'
+
+const props = defineProps<{
+  elementToDelete: 'board' | 'task'
+  elementName: string
```
Comment (aleksanderwalczuk) [inline src/components/Dialogs/ConfirmationDialog.vue:32]: I would rename it to `name` and `type`

File: src/components/Dialogs/DialogsTemplate.vue [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+const props = defineProps<{
+  closeDialog: () => void
+}>()
+
+const closeDialogOnEsc = (e: KeyboardEvent) => {
+  if (e.key === 'Escape') {
+    props.closeDialog()
+  }
+}
+window.addEventListener('keydown', (e: KeyboardEvent) => closeDialogOnEsc(e))
```
Comment (aleksanderwalczuk) [inline src/components/Dialogs/DialogsTemplate.vue]: Add when mounted, remove on unMount

File: src/components/Dialogs/SeeTaskDialog.vue [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+    <template #main-content>
+      <div class="flex flex-col gap-6 relative">
+        <more-options
+          :condition="areTaskOptionsShown"
+          element="task"
+          :toggleOptions="(e: Event) => callMoreOptionsFn(e, toggleOptions)"
+          :closeMoreOptions="(e: Event) => callMoreOptionsFn(e, closeOptions)"
+          :showEditForm="showEditForm"
+          :showDeleteForm="showDeleteForm"
+          :closeSeeTask="closeSeeTask"
```
Comment (aleksanderwalczuk) [inline src/components/Dialogs/SeeTaskDialog.vue]: Use emits

File: src/components/Dialogs/SeeTaskDialog.vue [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+        <p class="text-gray-400 text-xs min-[350px]:text-sm">
+          {{ boardProperties.description }}
+        </p>
+
+        <div>
+          <p class="mb-4 text-xs text-gray-400 dark:text-white">
+            Subtasks (
+            {{ returnNumberOfCompletedSubtasks(boardProperties.subtasks) }}
+            of
+            {{ boardProperties.subtasks.length }}
```
Comment (aleksanderwalczuk) [inline src/components/Dialogs/SeeTaskDialog.vue]: What if it's equal to `0`?

File: src/components/Dialogs/SeeTaskDialog.vue [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+            )
+          </p>
+          <div
+            v-for="subtask in boardProperties.subtasks"
```
Comment (aleksanderwalczuk) [inline src/components/Dialogs/SeeTaskDialog.vue]: Use destructuring

File: src/components/Dialogs/SeeTaskDialog.vue [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+}>()
+
+const boardProperties = returnBoardProperties()
+const statusItems = boardProperties.columns.map((column) => column.name)
+const areTaskOptionsShown = ref(false)
+
+const { toggleOptions, closeOptions } = moreOptionsPopup
+const callMoreOptionsFn = (
+  e: Event,
+  cb: (e: Event, conditionToChange: Ref<boolean>) => void
```
Comment (aleksanderwalczuk) [inline src/components/Dialogs/SeeTaskDialog.vue:104]: Why do you pass a ref instead of its value?

Comment (aleksanderwalczuk) [reply src/components/Dialogs/SeeTaskDialog.vue:104]: Avoid data mutation outside it's component

File: src/components/Dialogs/elements/MultiOption.vue [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+          :name="item"
+          :error="isError"
+          :class="{ 'after:content-none': !isError }"
+          class="subtask-input"
+        />
+        <svg
+          width="15"
+          height="15"
+          xmlns="http://www.w3.org/2000/svg"
+          class="cursor-pointer"
```
Comment (aleksanderwalczuk) [inline src/components/Dialogs/elements/MultiOption.vue]: Why pointer class?

File: src/components/EmptyInfo.vue [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+    v-if="emptyDashboard || emptyBoard"
+    class="flex flex-col items-center justify-center"
+  >
+    <p
+      class="mb-[25px] text-lg text-center text-gray-400 first-letter:capitalize"
+    >
+      {{ emptyInfoSpans.spanOne }}
+      is empty. Create a new
+      {{ emptyInfoSpans.spanTwo }}
+      to get started.
```
Comment (aleksanderwalczuk) [inline src/components/EmptyInfo.vue]: Whole msg should be computed value

File: src/components/EmptyInfo.vue [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+</template>
+
+<script setup lang="ts">
+import TheButton from '../components/shared/TheButton.vue'
+import { returnBoardProperties } from '../composables/boardProperties'
+import { computed } from 'vue'
+
+const boardProperties = returnBoardProperties()
+const emptyDashboard = computed(() => boardProperties.boards.length === 0)
+const emptyBoard = computed(() => boardProperties.columns.length === 0)
```
Comment (aleksanderwalczuk) [inline src/components/EmptyInfo.vue]: No reason for separate getters

File: src/components/Layouts/AuthLayout.vue
Comment (aleksanderwalczuk) [inline src/components/Layouts/AuthLayout.vue:1]: Should be split in two files as it's two separate contexts

File: src/components/Layouts/PrivacyPolicyLayout.vue [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+  </main>
+</template>
+
+<script setup lang="ts">
+import CloseIcon from '../Svgs/CloseIcon.vue'
+import PrivacyPolicyContent from '../PrivacyPolicyContent.vue'
+import { useRoute } from 'vue-router'
+
+defineProps<{
+  closePopup?: () => void
```
Comment (aleksanderwalczuk) [inline src/components/Layouts/PrivacyPolicyLayout.vue]: Emit

File: src/components/Navbar/BoardLabel.vue [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+}>()
+</script>
+
+<style scoped>
+.board-label {
+  @apply flex items-center gap-2 min-[350px]:gap-4 h-12 w-[90%] px-3 min-[350px]:px-6;
+  @apply rounded-r-[100px] cursor-pointer sm:pr-11;
+  @apply hover:bg-purple-320 hover:text-purple-400 hover:fill-purple-400;
+  @apply focus:bg-purple-320 focus:text-purple-400 focus:fill-purple-400;
+  @apply dark:hover:bg-white dark:focus:bg-white transition-colors duration-300;
```
Comment (aleksanderwalczuk) [inline src/components/Navbar/BoardLabel.vue:37]: Extra space

File: src/components/Navbar/BoardsNavbar.vue [hunk 15]
```diff
@@ -0,0 +1,118 @@
...
+        'absolute sm:scale-0': width < 640,
+        'scale-0 sm:scale-100': width >= 640
+      }"
+    >
+      <div v-if="width >= 640" class="px-[10%] mt-4 mb-[54px]">
+        <img
+          :src="theme ? '/img/logo-light.svg' : '/img/logo-dark.svg'"
+          alt="app logo"
+          class="h-[26px] w-[153px]"
+        />
```
Comment (aleksanderwalczuk) [inline src/components/Navbar/BoardsNavbar.vue]: Maybe you could use color and fill properties and style one image with css?

File: src/components/Navbar/BoardsNavbar.vue [hunk 16]
```diff
@@ -0,0 +1,118 @@
...
+          />
+        </div>
+      </div>
+    </nav>
+  </transition>
+  <transition name="dialog">
+    <board-dialog
+      v-if="isAddBoardDialogShown"
+      action="add"
+      :closeDialog="() => (isAddBoardDialogShown = false)"
```
Comment (aleksanderwalczuk) [inline src/components/Navbar/BoardsNavbar.vue]: Emit

File: src/components/Navbar/BoardsNavbar.vue [hunk 17]
```diff
@@ -0,0 +1,118 @@
...
+import BoardDialog from '../Dialogs/BoardDialog.vue'
+import { ref } from 'vue'
+
+defineProps<{
+  condition: boolean
+  width: number
+  theme: boolean
+  boards: Board[]
+  boardName: Board['name']
+  toggleSidebar: () => void
```
Comment (aleksanderwalczuk) [inline src/components/Navbar/BoardsNavbar.vue]: Emit

File: src/components/Navbar/MainNavbar.vue [hunk 18]
```diff
@@ -0,0 +1,167 @@
+<template>
+  <div class="grid" :class="{ 'col-span-2': isLogo }">
+    <more-options
+      :condition="areBoardOptionsShown"
+      element="board"
+      :showEditForm="() => (isEditBoardDialogShown = true)"
+      :showDeleteForm="() => (isDeleteBoardDialogShown = true)"
+      :toggleOptions="(e: Event) => callMoreOptionsFn(e, toggleOptions)"
+      :closeMoreOptions="(e: Event) => callMoreOptionsFn(e, closeOptions)"
```
Comment (aleksanderwalczuk) [inline src/components/Navbar/MainNavbar.vue]: Emits

File: src/components/Navbar/MainNavbar.vue [hunk 19]
```diff
@@ -0,0 +1,167 @@
...
+    </nav>
+    <transition name="dialog">
+      <task-dialog
+        v-if="isAddTaskDialogShown"
+        action="add"
+        :selectedMultiOptionItems="[
+          'e.g. Make coffee',
+          'e.g. Drink coffee & smile'
+        ]"
+        :closeDialog="() => (isAddTaskDialogShown = false)"
```
Comment (aleksanderwalczuk) [inline src/components/Navbar/MainNavbar.vue]: Emit

File: src/components/Navbar/MainNavbar.vue [hunk 20]
```diff
@@ -0,0 +1,167 @@
...
+      />
+    </transition>
+    <transition name="dialog">
+      <confirmation-dialog
+        v-if="isDeleteBoardDialogShown"
+        elementToDelete="board"
+        elementName="Platform Launch"
+        :closeDialog="() => (isDeleteBoardDialogShown = false)"
```
Comment (aleksanderwalczuk) [inline src/components/Navbar/MainNavbar.vue]: Emit

File: src/components/Navbar/MainNavbar.vue [hunk 21]
```diff
@@ -0,0 +1,167 @@
...
+      />
+    </transition>
+    <transition name="dialog">
+      <board-dialog
+        v-if="isEditBoardDialogShown"
+        action="edit"
+        :closeDialog="() => (isEditBoardDialogShown = false)"
```
Comment (aleksanderwalczuk) [inline src/components/Navbar/MainNavbar.vue]: Emkt

File: src/components/TaskCard.vue [hunk 22]
```diff
@@ -0,0 +1,36 @@
...
+    </p>
+  </div>
+</template>
+
+<script setup lang="ts">
+defineProps<{
+  title: string
+  howManyCompleted: number
+  howManySubtasks: number
+  showTaskDetails: (e: Event) => void
```
Comment (aleksanderwalczuk) [inline src/components/TaskCard.vue]: Emit

File: src/components/shared/InputText.vue [hunk 23]
```diff
@@ -0,0 +1,60 @@
...
+      v-if="type === 'text'"
+      :type="type"
+      class="input"
+      :class="{
+        'border-red-400': error,
+        'border-blue-40 focus:border-purple-400': !error
+      }"
+      :placeholder="placeholder"
+      :value="customValue ? customValue : ''"
+    />
```
Comment (aleksanderwalczuk) [inline src/components/shared/InputText.vue:14]: What for?

File: src/components/shared/InputText.vue [hunk 24]
```diff
@@ -0,0 +1,60 @@
...
+        'border-blue-40 focus:border-purple-400': !errorMessage
+      }"
+    />
+    <ErrorMessage :name="name" class="font-normal text-xs text-red-400" />
+    <textarea
+      v-if="type === 'textarea'"
+      class="input min-h-[80px] max-h-[150px] scrollbar-invisible"
+      :class="{ 'border-blue-40 focus:border-purple-400': !error }"
+      placeholder="e.g. It’s always good to take a break. This 15 minute break will recharge the batteries a little."
+    ></textarea>
```
Comment (aleksanderwalczuk) [inline src/components/shared/InputText.vue:33]: Why it's not just a field?

File: src/components/shared/InputText.vue [hunk 25]
```diff
@@ -0,0 +1,60 @@
...
+import { Field, ErrorMessage, useField } from 'vee-validate'
+
+const props = defineProps<{
+  label?: string
+  placeholder?: string
+  customValue?: string
+  error?: boolean
+  formType?: 'add' | 'edit'
+  name: string
+  type: 'text' | 'email' | 'password' | 'textarea'
```
Comment (aleksanderwalczuk) [inline src/components/shared/InputText.vue]:
```suggestion
  type: HTMLInputElement['type']
```

File: src/components/shared/MoreOptions.vue [hunk 26]
```diff
@@ -0,0 +1,69 @@
...
+import { onClickOutside } from '@vueuse/core'
+import { ref } from 'vue'
+
+const props = defineProps<{
+  condition: boolean
+  element: 'task' | 'board'
+  showEditForm: () => void
+  showDeleteForm: () => void
+  toggleOptions: (e: Event) => void
+  closeMoreOptions: (e: Event) => void
```
Comment (aleksanderwalczuk) [inline src/components/shared/MoreOptions.vue]: Emits

File: src/components/shared/TheButton.vue [hunk 27]
```diff
@@ -0,0 +1,18 @@
+<template>
+  <button
+    :class="{
+      'flex items-center grow justify-center gap-2 py-[10px] px-4 rounded-3xl':
+        regularButton,
+      'text-sm': isInForm
```
Comment (aleksanderwalczuk) [inline src/components/shared/TheButton.vue:6]: This class could be applied inline

File: src/components/shared/ThemeToggle.vue [hunk 28]
```diff
@@ -0,0 +1,37 @@
+<template>
+  <div class="flex items-center justify-center gap-x-6 mx-auto h-12">
+    <img src="/img/icon-light-theme.svg" alt="" />
+    <the-button
+      @click="toggleDark()"
```
Comment (aleksanderwalczuk) [inline src/components/shared/ThemeToggle.vue]: This should be a function reference

File: src/composables/boardProperties.ts
Comment (aleksanderwalczuk) [inline src/composables/boardProperties.ts:1]: What it does and why it's outside of the store?

File: src/composables/completedTasks.ts
Comment (aleksanderwalczuk) [inline src/composables/completedTasks.ts:1]: One-liner, no need for separate file

File: src/composables/completedTasks.ts [hunk 29]
```diff
@@ -0,0 +1,4 @@
+import type { Subtask } from '../api/boardsTypes'
+
+export const returnNumberOfCompletedSubtasks = (arr: Subtask[]) =>
```
Comment (aleksanderwalczuk) [inline src/composables/completedTasks.ts]: One-liner, no need for separate file

File: src/composables/moreOptionsPopup.ts
Comment (aleksanderwalczuk) [inline src/composables/moreOptionsPopup.ts:1]: This is bad. It relies on side effects

File: src/stores/boards.ts [hunk 30]
```diff
@@ -0,0 +1,451 @@
+import { defineStore } from 'pinia'
+import { ref } from 'vue'
+import type { Ref } from 'vue'
+import type { Board } from '../api/boardsTypes'
+
+export const useBoardsStore = defineStore('boards', () => {
+  const boards: Ref<Board[]> = ref([
```
Comment (aleksanderwalczuk) [inline src/stores/boards.ts:7]: Not DRY

File: src/stores/user.ts [hunk 31]
```diff
@@ -1,10 +1,55 @@
...
+    }
+  }
+
+  const logOutUser = async () => {
+    try {
+      await signOut(auth)
+      return true
+    } catch (err) {
+      return false
+    }
```
Comment (aleksanderwalczuk) [inline src/stores/user.ts]: Clear user on logout

File: src/stores/user.ts [hunk 32]
```diff
@@ -1,10 +1,55 @@
...
+        await logOutUser()
+      }
+
+      return true
+    } catch (err) {
+      return false
+    }
+  }
+
+  const logOutUser = async () => {
```
Comment (aleksanderwalczuk) [inline src/stores/user.ts]: Change name to logout

File: src/views/LoginView.vue
Comment (aleksanderwalczuk) [inline src/views/LoginView.vue:1]: Pointless component

File: src/views/SignUpView.vue
Comment (aleksanderwalczuk) [inline src/views/SignUpView.vue:1]: Pointless component

**Review by aleksanderwalczuk — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (Kepmon) [reply package.json]: **Done**

File: src/components/BoardsColumn.vue (hunk 2 above)
Comment (Kepmon) [reply src/components/BoardsColumn.vue]: I guess, it's done but I'm not really sure about my solution being right. 

In my child component, I have the code like this:
```js
<template>
  <div
    @click="$emit('changeTaskTitle', title)"
    tabindex="0"
    class="task-card group"
  >
    <p class="task-card__title">
      {{ title }}
    </p>
    <p class="text-gray-400 text-xs">
      {{ howManyCompleted }} of {{ howManySubtasks }} subtasks
    </p>
  </div>
</template>

<script setup lang="ts">
defineProps<{
  title: string
  howManyCompleted: number
  howManySubtasks: number
}>()
defineEmits(['changeTaskTitle'])
</script>
```

but according to the Vue docs ([here](https://vuejs.org/guide/components/events.html#event-arguments)), I don't really need this `defineEmits` thing. I added it because eslint yelled at me I should do so, but I don't think I understand their exaplanation [here](https://eslint.vuejs.org/rules/require-explicit-emits.html) on why this is the correct approach.

Also, I'd like to emphasize here, this part of the code will be changed in the future anyway, because, currently, all related data is hardcoded.

Comment (aleksanderwalczuk) [reply src/components/BoardsColumn.vue]: Why not just `change`?

File: src/components/Dialogs/BoardDialog.vue (hunk 3 above)
Comment (Kepmon) [reply src/components/Dialogs/BoardDialog.vue]: **Done**

File: src/components/Navbar/BoardsNavbar.vue (hunk 16 above)
Comment (Kepmon) [reply src/components/Navbar/BoardsNavbar.vue]: **Done**

File: src/components/Navbar/MainNavbar.vue (hunk 19 above)
Comment (Kepmon) [reply src/components/Navbar/MainNavbar.vue]: **Done**

File: src/components/Navbar/MainNavbar.vue (hunk 20 above)
Comment (Kepmon) [reply src/components/Navbar/MainNavbar.vue]: **Done**

File: src/components/Navbar/MainNavbar.vue (hunk 21 above)
Comment (Kepmon) [reply src/components/Navbar/MainNavbar.vue]: **Done**

File: src/components/BoardsColumn.vue (hunk 2 above)
Comment (Kepmon) [reply src/components/BoardsColumn.vue]: Because, recently, I've liked this idea of naming things (functions, variables) the most descriptive way possible. If I was to come back to this code in a few weeks, let say, I would have no idea what `change` means and what's supposed to be changed. Whereas, `changeTaskTitle` seems to be pretty self-descriptive to me.

And I believe we both know it's highly possible I will be coming back to this code many times in the not-so-distant future, so I'd better know what my functions do 🙃

File: src/components/Dialogs/ConfirmationDialog.vue (hunk 4 above)
Comment (Kepmon) [reply src/components/Dialogs/ConfirmationDialog.vue:32]: [Same answer](https://github.com/Kepmon/task-manager/pull/3#discussion_r1240522871)

File: src/components/Dialogs/DialogsTemplate.vue (hunk 5 above)
Comment (Kepmon) [reply src/components/Dialogs/DialogsTemplate.vue]: **Done**

File: src/components/Dialogs/SeeTaskDialog.vue (hunk 7 above)
Comment (Kepmon) [reply src/components/Dialogs/SeeTaskDialog.vue]: Good question 🙃

**Corrected**, added `v-if` to both components (this one and `TaskCard.vue`) that display the number of the subtasks completed.

File: src/components/Dialogs/SeeTaskDialog.vue (hunk 8 above)
Comment (Kepmon) [reply src/components/Dialogs/SeeTaskDialog.vue]: **Done**

File: src/components/Dialogs/elements/MultiOption.vue (hunk 10 above)
Comment (Kepmon) [reply src/components/Dialogs/elements/MultiOption.vue]: Because those are the closing icons:
add_form

But you're right, it should be a button along with the respective aria-label and some padding around it 😛 
**Corrected**

File: src/components/shared/InputText.vue (hunk 23 above)
Comment (Kepmon) [reply src/components/shared/InputText.vue:14]: Because I have this input component in other forms as well (I mean, besides the auth forms) and in those forms they're of type text and as for now - they're not supposed to be validated. I may add some validation so they can't be empty in the future but even if so - I will have to add a separate `Yup` thing to be able to do that. Also, the "Can't be empty" error will be displayed in those as a placeholder/value since that's what my mockup implies.

File: src/components/shared/InputText.vue (hunk 24 above)
Comment (Kepmon) [reply src/components/shared/InputText.vue:33]: What field?

File: src/components/shared/InputText.vue (hunk 25 above)
Comment (Kepmon) [reply src/components/shared/InputText.vue]: **Done**

File: src/components/EmptyInfo.vue (hunk 11 above)
Comment (Kepmon) [reply src/components/EmptyInfo.vue]: **Done**

File: src/components/EmptyInfo.vue (hunk 12 above)
Comment (Kepmon) [reply src/components/EmptyInfo.vue]: **Done**, but not sure if it's the fix you had in mind.

File: src/components/Layouts/PrivacyPolicyLayout.vue (hunk 13 above)
Comment (Kepmon) [reply src/components/Layouts/PrivacyPolicyLayout.vue]: **Done**

File: src/components/Navbar/BoardLabel.vue (hunk 14 above)
Comment (Kepmon) [reply src/components/Navbar/BoardLabel.vue:37]: Not sure what you mean?

If this is about this empty space after `sm:pr-11`, it's because I like to group the Tailwind classess (when being separated for several lines like here), so each of the lines contains similar/related properties. As such, when I need to change something, e.g. the hover and focus states, I know all hover-related classess are in one line and the focus-related classess are in the other line and I don't have to read everything to be able to find what I'm after.

File: src/components/Navbar/BoardsNavbar.vue (hunk 15 above)
Comment (Kepmon) [reply src/components/Navbar/BoardsNavbar.vue]: **Done**

File: src/components/Navbar/BoardsNavbar.vue (hunk 17 above)
Comment (Kepmon) [reply src/components/Navbar/BoardsNavbar.vue]: **Done**

File: src/components/Navbar/MainNavbar.vue (hunk 18 above)
Comment (Kepmon) [reply src/components/Navbar/MainNavbar.vue]: **Done**

File: src/components/TaskCard.vue (hunk 22 above)
Comment (Kepmon) [reply src/components/TaskCard.vue]: **Done**

File: src/components/shared/MoreOptions.vue (hunk 26 above)
Comment (Kepmon) [reply src/components/shared/MoreOptions.vue]: **Done**

File: src/components/shared/TheButton.vue (hunk 27 above)
Comment (Kepmon) [reply src/components/shared/TheButton.vue:6]: Which one?

Because, if you mean the `text-sm` one, I'm afraid prettier doesn't agree with you. Unless you want me to disable prettier for this line?

File: src/components/shared/ThemeToggle.vue (hunk 28 above)
Comment (Kepmon) [reply src/components/shared/ThemeToggle.vue]: **Done**

File: src/composables/boardProperties.ts
Comment (Kepmon) [reply src/composables/boardProperties.ts:1]: It's a temporary file for destructuring data I have hardcoded in the current version of my code.

However, it should be in the store, **corrected**.

File: src/composables/completedTasks.ts
Comment (Kepmon) [reply src/composables/completedTasks.ts:1]: **Moved it to the store**

File: src/composables/completedTasks.ts (hunk 29 above)
Comment (Kepmon) [reply src/composables/completedTasks.ts]: **Moved it to the store**

File: src/composables/moreOptionsPopup.ts
Comment (Kepmon) [reply src/composables/moreOptionsPopup.ts:1]: I don't understand - what side effects?

File: src/stores/boards.ts (hunk 30 above)
Comment (Kepmon) [reply src/stores/boards.ts:7]: I don't know where exactly I repeat myself in this code but this data is temporarily stored here. Eventually, this data will vanish and all user data will be handled through firebase.

File: src/stores/user.ts (hunk 31 above)
Comment (Kepmon) [reply src/stores/user.ts]: Yeah, I forgot about that. **Done**.

File: src/stores/user.ts (hunk 32 above)
Comment (Kepmon) [reply src/stores/user.ts]: **Done**

File: src/views/LoginView.vue
Comment (Kepmon) [reply src/views/LoginView.vue:1]: Hmm... I believe I need it for routing.

File: src/views/SignUpView.vue
Comment (Kepmon) [reply src/views/SignUpView.vue:1]: [Same answer](https://github.com/Kepmon/task-manager/pull/3#discussion_r1240957350)

File: src/components/Dialogs/SeeTaskDialog.vue (hunk 6 above)
Comment (Kepmon) [reply src/components/Dialogs/SeeTaskDialog.vue]: **Done**

File: src/components/Layouts/AuthLayout.vue
Comment (Kepmon) [reply src/components/Layouts/AuthLayout.vue:1]: Unfortunately, I can't agree on this one.

File: src/components/Dialogs/SeeTaskDialog.vue (hunk 9 above)
Comment (Kepmon) [reply src/components/Dialogs/SeeTaskDialog.vue:104]: > Why do you pass a ref instead of its value?

Not sure (when doing that I was just trying different things until something worked) but I can see now, the function doesn't work when I pass `areTaskOptionsShown.value` instead of `areTaskOptionsShown`. I believe, it's because the callback is ultimately being executed in the template, where `areTaskOptionsShown.value` doesn't exist. But I may be wrong on this one. Still, passing a ref to that seems to me to be the only way to go.


> Avoid data mutation outside it's component

You mean, I shouldn't have those helper functions? I did this because I actually need them in two separate component and I didn't want to violate this DRY thing you mentioned in one of your following comments...

File: src/components/BoardsColumn.vue [hunk 33]
```diff
@@ -0,0 +1,122 @@
...
+          <div
+            class="h-[15px] w-[15px] rounded-full"
+            :class="circleColor(column)"
+          ></div>
+          <p class="text-xs text-gray-400 uppercase">
+            {{ column.name }} ({{ column.tasks.length }})
+          </p>
+        </div>
+        <task-card
+          @change-task-title="(title) => (clickedTitle = title)"
```
Comment (aleksanderwalczuk) [inline src/components/BoardsColumn.vue]: This is wrong. Task should have a unique `id`. On `@change-title` this id should be emitted to the parent. With find method you should get that task object and change it's title within the data object.

File: src/components/BoardsColumn.vue (hunk 2 above)
Comment (aleksanderwalczuk) [reply src/components/BoardsColumn.vue]: Of course, but being descriptive means also being flexible. What if during next iteration you'd like to edit the `description` too? 
Would you create another function with another emit? This is against DRY, isn't it? You should also be familiar with the KISS rule. So in my opinion `changeTask` could receive the `title` parameter. It's still descriptive, but if you'd like to change the parameter it's also easy, right?

File: src/components/Dialogs/BoardDialog.vue [hunk 34]
```diff
@@ -0,0 +1,46 @@
+<template>
+  <dialogs-template @close-dialog="$emit('close-dialog')">
```
Comment (aleksanderwalczuk) [inline src/components/Dialogs/BoardDialog.vue:2]: Just `close` would be enough

File: src/components/Dialogs/ConfirmationDialog.vue (hunk 4 above)
Comment (aleksanderwalczuk) [reply src/components/Dialogs/ConfirmationDialog.vue:32]: Are you sure that the thing you want to delete is an `element`?

Comment (aleksanderwalczuk) [reply src/components/Dialogs/ConfirmationDialog.vue:32]: Beside that there's nothing else happening inside this component

**Review by aleksanderwalczuk — CHANGES_REQUESTED** [review]

File: src/components/BoardsColumn.vue (hunk 2 above)
Comment (Kepmon) [reply src/components/BoardsColumn.vue]: Yeah... so, the whole function that edits the task title looks like this:
```js
@change-task-title="(title) => (clickedTitle = title)"
```

So, if I wanted to change `description` as well, I would have to repeat myself anyway.

I wasn't familiar with the KISS rule but "simple" to me would mean descriptive enough so I know what it does without reading the whole code.

So, not really conviced on this one, but since you insist - I **changed it**.

File: src/components/Dialogs/ConfirmationDialog.vue (hunk 4 above)
Comment (Kepmon) [reply src/components/Dialogs/ConfirmationDialog.vue:32]: > Are you sure that the thing you want to delete is an element?

To me it's actually just a `thing` but I don't find it to be an appropriate name, so I went with the `element` name. What would you suggest instead?

> Beside that there's nothing else happening inside this component

I'm afraid I don't understand what do you mean by that. The component renders the "Are you sure you want to delete this thing" dialog... what do you expect for it to do?

File: src/components/BoardsColumn.vue (hunk 33 above)
Comment (Kepmon) [reply src/components/BoardsColumn.vue]: This function doesn't really change the task title, it changes the title of the clicked task. But it doesn't really matter because this is the part of my hardcoded solutions. I don't even remember why I put this function there in the first place, probably wanted to have some starting point for when I actually do have some real tasks with id.

So, until I store all user data in firebase, there is nothing really that I can do about that (beside deleting this whole function) but I'll remember your suggestion for later on.

File: src/components/Dialogs/BoardDialog.vue (hunk 34 above)
Comment (Kepmon) [reply src/components/Dialogs/BoardDialog.vue:2]: Yeah, but I strongly believe `close-dialog` is as simple as `close`, concurrently being more descriptive. And again - all related functions are one-liners, written in the template.

File: src/components/Dialogs/SeeTaskDialog.vue (hunk 9 above)
Comment (aleksanderwalczuk) [reply src/components/Dialogs/SeeTaskDialog.vue:104]: You shouldn't pass the ref and update it's value inside a component in which it isn't registered. This makes code hard to follow. Use the value instead. I'll create a separate PR to fix this, you'll merge it or you won't :)

File: src/components/Layouts/AuthLayout.vue
Comment (aleksanderwalczuk) [reply src/components/Layouts/AuthLayout.vue:1]: Then your code will be messy and harder to maintain :)

File: src/components/Layouts/AuthLayout.vue [hunk 35]
```diff
@@ -129,36 +140,26 @@ const form = useForm({
...
-    isAuthError.value = true
-  }
-
-  if (isUserAuthenticated) {
-    isAuthError.value = false
-    setTimeout(() => {
-      router.push(`${currentPath === '/sign-up' ? '/' : '/dashboard'}`)
-    }, 2000)
+  if (response !== true) {
+    errorMessage.value = response
```
Comment (aleksanderwalczuk) [inline src/components/Layouts/AuthLayout.vue:159]: So the response is either a boolean or an object? 😮

File: src/components/Navbar/BoardLabel.vue (hunk 14 above)
Comment (aleksanderwalczuk) [reply src/components/Navbar/BoardLabel.vue:37]: Extra space means either your indentation is wrong or there are two spaces between blocks of text and one should be deleted. I think it's ok now anyway.

File: src/components/BoardsColumn.vue [hunk 36]
```diff
@@ -1,5 +1,9 @@
 <template>
-  <div class="columns-container" :class="{ 'columns-container--sizes': !logo }">
+  <div
+    v-if="columns"
```
Comment (aleksanderwalczuk) [inline src/components/BoardsColumn.vue:3]: what for? An array is truthy, right?

File: src/components/Navbar/BoardsNavbar.vue [hunk 37]
```diff
@@ -66,22 +80,28 @@ import type { Board } from '../../api/boardsTypes'
...
 import { ref } from 'vue'
 
 defineProps<{
   condition: boolean
   width: number
   theme: boolean
-  boards: Board[]
-  boardName: Board['name']
+  boards: Board[] | null
+  boardName: Board['name'] | ''
```
Comment (aleksanderwalczuk) [inline src/components/Navbar/BoardsNavbar.vue:91]: why an empty string?

Comment (aleksanderwalczuk) [reply src/components/Navbar/BoardsNavbar.vue:91]: as a literal... 🤔

File: src/components/Navbar/MainNavbar.vue [hunk 38]
```diff
@@ -78,6 +78,10 @@
           </svg>
         </the-button>
       </div>
+      <Teleport v-if="width < 512" to="body">
+        <user-options class="absolute bottom-8 right-8 scale-125" />
+      </Teleport>
+      <user-options v-else class="static ml-2 sm:ml-3" />
```
Comment (aleksanderwalczuk) [inline src/components/Navbar/MainNavbar.vue:84]: Does it work on window resize?

File: src/firebase.ts [hunk 39]
```diff
@@ -14,3 +15,5 @@ const firebaseConfig = {
 
 export const app = initializeApp(firebaseConfig)
 export const auth = getAuth()
+const db = getFirestore()
+export const colRef = collection(db, 'users')
```
Comment (aleksanderwalczuk) [inline src/firebase.ts:19]: what is it?

File: src/stores/boardsNew.ts [hunk 40]
```diff
@@ -0,0 +1,34 @@
...
+
+  const returnNumberOfCompletedSubtasks = (arr: Subtask[]) =>
+    arr.filter((subtask) => subtask.isCompleted === true).length
+
+  return {
+    getBoardsData,
+    boards,
+    returnNumberOfCompletedSubtasks
+  }
+})
```
Comment (aleksanderwalczuk) [inline src/stores/boardsNew.ts:34]: I'll have to get back to this one.

**Comment by Kepmon** [comment]
I believe there were so many changes (both mine and yours) since I opened this PR that there is no point on discussing this right now. Since you allowed me to merge this branch, I'll do it, and if any of your question is still relevant during our live refactoring, we'll discuss it then.
